### PR TITLE
[enhance](mtmv)allow add index for MTMV

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

Previously, the limitation on whether operations can be performed on materialized views was to determine `opType`.

Now, a `allowOpMTMV()` method is implemented through various `clauses`.

Because some operations have the same `opType`, but some operations allow and some do not.

For example, the `opType` for both `add column` and `create index` is `SCHEMA-CHANGE`, but `add column` is not allowed and `create index` is allowed.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AddRollupClause.java [hunk 1]
```diff
@@ -89,6 +89,11 @@ public void analyze(Analyzer analyzer) throws AnalysisException {
         baseRollupName = Strings.emptyToNull(baseRollupName);
     }
 
+    @Override
+    public boolean allowOpMTMV() {
+        return false;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AddRollupClause.java:94]: why cannot add rollup?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnRenameClause.java [hunk 2]
```diff
@@ -63,6 +63,11 @@ public Map<String, String> getProperties() {
         return null;
     }
 
+    @Override
+    public boolean allowOpMTMV() {
+        return false;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnRenameClause.java:68]: why cannot rename?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ModifyColumnCommentClause.java [hunk 3]
```diff
@@ -59,6 +59,11 @@ public void analyze(Analyzer analyzer) throws AnalysisException {
         }
     }
 
+    @Override
+    public boolean allowOpMTMV() {
+        return false;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ModifyColumnCommentClause.java:64]: why？

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ReorderColumnsClause.java [hunk 4]
```diff
@@ -67,6 +67,11 @@ public Map<String, String> getProperties() {
         return this.properties;
     }
 
+    @Override
+    public boolean allowOpMTMV() {
+        return false;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ReorderColumnsClause.java:72]: why?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/RollupRenameClause.java [hunk 5]
```diff
@@ -63,6 +63,11 @@ public Map<String, String> getProperties() {
         return null;
     }
 
+    @Override
+    public boolean allowOpMTMV() {
+        return false;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/RollupRenameClause.java:68]: why?

**Review by morrySnow — DISMISSED** [review]

**Comment by zddr** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by seawinde — APPROVED** [review]

**Pull request merged**